gnomad_qc/v4/resources/basics.py

```diff
@@ -119,28 +121,40 @@ def _remove_ukb_dup_by_index(
         "Total number of UKB samples removed from the VDS: %d", n_samples_removed
     )
 
-    if remove_hard_filtered_samples or remove_hard_filtered_samples_no_sex:
+    if (
+        high_quality_only
+        or remove_hard_filtered_samples
+        or remove_hard_filtered_samples_no_sex
+    ):
         if test:
             meta_ht = gnomad_v4_testset_meta.ht()
-            if remove_hard_filtered_samples_no_sex:
-                hard_filter_expr = meta_ht.rand_sampling_meta.hard_filters_no_sex
+            if high_quality_only:
+                filter_expr = meta_ht.high_quality
+            elif remove_hard_filtered_samples_no_sex:
+                filter_expr = (
+                    hl.len(meta_ht.rand_sampling_meta.hard_filters_no_sex) == 0
+                )
             else:
-                hard_filter_expr = meta_ht.rand_sampling_meta.hard_filters
-            meta_ht = meta_ht.filter(hl.len(hard_filter_expr) == 0)
+                filter_expr = hl.len(meta_ht.rand_sampling_meta.hard_filters) == 0
+            meta_ht = meta_ht.filter(filter_expr)
             vds = hl.vds.filter_samples(vds, meta_ht)
         else:
             from gnomad_qc.v4.resources.sample_qc import (
+                finalized_outlier_filtering,
                 hard_filtered_samples,
                 hard_filtered_samples_no_sex,
             )
 
-            if remove_hard_filtered_samples:
-                hard_filter_ht = hard_filtered_samples.versions[CURRENT_VERSION].ht()
+            keep_samples = False
+            if high_quality_only:
+                filter_ht = finalized_outlier_filtering().ht()
+                filter_ht = filter_ht.filter(~filter_ht.outlier_filtered)
+                keep_samples = True
```

Comment:
Could you remind me why we can't do 
```suggestion
                filter_ht = filter_ht.filter(filter_ht.outlier_filtered)
```
Is it because we may have missing values and those would be dropped?

Reply:
The `finalized_outlier_filtering` already has hard-filtered and unreleasable samples removed from it, so doing it with a keep_samples = False would also keep them